app/kuma-ui/pkg/resources/data/index.html

```diff
@@ -1,15 +1,18 @@
 <!DOCTYPE html>
 <html lang="en">
   <head>
+    <script id="kuma-config" type="application/json">
+      {{.}}
+    </script>
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width,initial-scale=1" />
 
-    <link rel="icon" href="./favicon.png" />
+    <link rel="icon" href="/{{ .BaseGuiPath | trimPrefix "/" }}/favicon.png" />
```

Comment:
**Issue**: I’ve tried generating a placeholder in this form and it won’t work.

The build script will output it with escaped quotation marks because it would otherwise make the attribute value invalid (again, it doesn’t know that this is subject to further templating).

```html
<script type="module" crossorigin src="/{{ .BaseGuiPath | trimPrefix \"/\" }}/assets/index.6b8c6d03.js"></script>
```

Can we please trim the value for `BaseGuiPath` beforehand? This would affect the JSON content, too, but that’s fine by me (there isn’t any ambiguity without the leading slash).

Reply:
Oh wait, do we need double quotes? The doubles get escaped there also 🤦 . Can go-template use singles?

Reply:
```javascript
function plugin(json) {
  return {
    name: 'transform-file',
    transformIndexHtml(html) {
      html = html.replaceAll(
        '/{{.BaseGuiPath trimPrefix \\"/\\"}}',
        '/{{.BaseGuiPath trimPrefix "/"}}',
      )
      if (json) {
        html = html.replaceAll(
          '{{.}}',
          JSON.stringify(json),
        )
      }
      return html
    },
  }
}
```

More or less that ^ as a plugin should work? Also conditionally pass in a piece of JSON for filling out the JSON during development.

Reply:
Can't use `'` in gotemplates no.

Isn't what Js mostly doing is post-processing HTML if we want to do it properly?
In any case just put `{{TP.BaseGuiPath}}` and then sed it. That'll be done once and forgotten about. Not having the apiPath rooted will cause more issues (Starting with folks on the backend looking at this and thinking "WTF is this this way?".
Feels like we can contain the complexity and weirdness to kuma-gui which seems to be the better option to me

Reply:
I don’t agree with the arguments brought forth for handling this problem in the frontend.

I acknowledge that our build tool has the unfortunate limitation that requires us to provide a path segment that has no leading or trailing slashes. That’s unfortunate because in principle, we’d want to only carry around base path value that _does_ start with a leading slash.

However, to reconcile this problem, we really only need to provide the path without a leading and trailing slash. This is technologically neither complex nor inaccurate or unclean in terms of what the definition of some sort of base path is. Calling it a base path already implies it is relative to a URL’s origin meaning that it’s unlikely that it’s going to be unclear in either backend or frontend application whether a concatenation with a slash will be required or not.

In contrast adding a post build step in the frontend to replace strings in an HTML is many times more complicated and also error-prone.

In summary, I don’t understand why it’s preferable to introduce the complexity in the frontend as proposed when the alternative is to process the index.html in the templating step such that its `BaseGuiPath` variable doesn’t contain a leading or trailing slash. I don’t see the problem with this approach.

---

On an aside, this does not at all affect the base API URL for which no changes are needed in the context of this conversation.

Reply:
It's all in this argument:

> Overall I think that we should code the frontend and the backend as if this little build tool problem didn't exist, and then fix it in the build tool, just for the specific build tool we are using.

@johncowen and I seems to be in agreement around this. Contain complexity where it should be. It's against the principle of least surprise to not have base path be rooted look for example at the swagger spec: https://swagger.io/docs/specification/2-0/api-host-and-base-path/